### ISSUE TITLE
panic: interface conversion: interface {} is nil, not map[string]interface aws_sagemaker_endpoint_configuration.async_inference_config.notification_config

### ISSUE THREAD

**Issue by jengel-rackner** [hashicorp/terraform-provider-aws#21633]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.7
AWS provider version ~> 4.0

### Affected Resource(s)

* aws_sagemaker_endpoint_configuration

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_sagemaker_endpoint_configuration" "ec" {
  count = var.sagemaker_data.endpoint_data == null ? 0 : 1

  name = "${var.name}-ec-${local.suffix}"

  dynamic "production_variants" {
    for_each = { for variant in var.sagemaker_data.endpoint_data.variants : variant.variant_name => variant }

    content {
      variant_name           = production_variants.key
      model_name             = aws_sagemaker_model.model[count.index].name
      initial_instance_count = production_variants.value["initial_instance_count"] == null ? 1 : production_variants.value["initial_instance_count"]
      instance_type          = production_variants.value["instance_type"]
    }
  }

  dynamic "async_inference_config" {
    for_each = var.sagemaker_data.endpoint_data.output_config == null ? [] : [var.sagemaker_data.endpoint_data.output_config]

    content {
      output_config {

        s3_output_path = "s3://${data.aws_s3_bucket.batch_transform_data.bucket}/${var.env}/${var.name}/outputs"

        dynamic "notification_config" {
          for_each = async_inference_config.value.success_topic != null || async_inference_config.value.error_topic != null ? [true] : []

          content {
            error_topic   = async_inference_config.value.error_topic != null ? module.resources.topics_hash_by_name[async_inference_config.value.error_topic].arn : null
            success_topic = async_inference_config.value.success_topic != null ? module.resources.topics_hash_by_name[async_inference_config.value.success_topic].arn : null
          }
        }
      }

      client_config {
        max_concurrent_invocations_per_instance = var.sagemaker_data.endpoint_data.max_concurrent_invocations_per_instance
      }

    }
  }

  dynamic "data_capture_config" {
    for_each = var.sagemaker_data.endpoint_data.data_capture_config == null ? [] : [var.sagemaker_data.endpoint_data.data_capture_config]

    content {
      initial_sampling_percentage = data_capture_config.value["initial_sampling_percentage"]
      destination_s3_uri          = "s3://${data.aws_s3_bucket.batch_transform_data.bucket}/${var.env}/${var.name}/outputs"
      enable_capture              = data_capture_config.value["enable_capture"]

      dynamic "capture_content_type_header" {
        for_each = data_capture_config.value["capture_content_type_header"] == null ? [] : [data_capture_config.value["capture_content_type_header"]]

        content {
          csv_content_types  = capture_content_type_header.value["csv_content_types"]
          json_content_types = capture_content_type_header.value["json_content_types"]
        }
      }

      dynamic "capture_options" {
        for_each = data_capture_config.value["capture_modes"]

        content {
          capture_mode = capture_options.value
        }
      }
    }
  }

  tags = local.tags
}
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output

https://gist.github.com/jengel-rackner/d2a157f8f66461a48ccfd01f63ac0b22

### Expected Behavior

Endpoint configuration deploys with both/either SNS error/success notification topics

### Actual Behavior

Received error

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by ewbankkit** [comment]
```
Stack trace from the terraform-provider-aws_v3.63.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 47 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.expandSagemakerEndpointConfigClientConfig(...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_sagemaker_endpoint_configuration.go:594
github.com/terraform-providers/terraform-provider-aws/aws.expandSagemakerEndpointConfigAsyncInferenceConfig(0xc0036f3a80, 0x1, 0x1, 0x69e9740)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_sagemaker_endpoint_configuration.go:579 +0x31f
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsSagemakerEndpointConfigurationCreate(0xc0036bc900, 0x74553e0, 0xc001832000, 0x5, 0xffffffffffffffff)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_sagemaker_endpoint_configuration.go:307 +0x398
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000fb4e00, 0x8f161a8, 0xc0036a2180, 0xc0036bc900, 0x74553e0, 0xc001832000, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/gdavison/terraform-plugin-sdk/v2@v2.7.1-0.20210913224932-c7c2dbd9e010/helper/schema/resource.go:318 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000fb4e00, 0x8f161a8, 0xc0036a2180, 0xc000d96460, 0xc0026267e0, 0x74553e0, 0xc001832000, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/gdavison/terraform-plugin-sdk/v2@v2.7.1-0.20210913224932-c7c2dbd9e010/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000c840, 0x8f161a8, 0xc0036a2180, 0xc0036a4370, 0xc0036a2180, 0x80d3a00, 0xc00369c100)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/gdavison/terraform-plugin-sdk/v2@v2.7.1-0.20210913224932-c7c2dbd9e010/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc001ce2820, 0x8f16250, 0xc0036a2180, 0xc000d96070, 0xc001ce2820, 0xc00369c1b0, 0xc003694ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x80d3a00, 0xc001ce2820, 0x8f16250, 0xc00369c1b0, 0xc00013c2a0, 0x0, 0x8f16250, 0xc00369c1b0, 0xc0036b5000, 0x7af)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000831880, 0x8f3a098, 0xc000354600, 0xc0036b0000, 0xc0004aa990, 0xc9c61c0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000831880, 0x8f3a098, 0xc000354600, 0xc0036b0000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00062ccd0, 0xc000831880, 0x8f3a098, 0xc000354600, 0xc0036b0000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-aws_v3.63.0_x5 plugin crashed!
```

**Comment by jaimi-taos-wellhealth** [comment]
I tried to use an older provider, but the issue still remains.

terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
      version = "~> 3.64.0"
    }
  }
  required_version = ">= 0.13"
}

**Comment by DrFaust92** [comment]
Should have been solved by #22960, if the issue persists using latest provider please open a new issue with details.
